python-avd/pyavd/_eos_designs/shared_utils/routing.py

```diff
@@ -109,19 +109,28 @@ def bgp_as_notation(self: SharedUtilsProtocol) -> Literal["asdot", "asplain"]:
     def get_asn(self: SharedUtilsProtocol, asn: str | int | None) -> str | None:
         if asn is None:
             return None
-
-        bgp_as_notation = self.bgp_as_notation
-        if bgp_as_notation == "asdot" and "." not in str(asn):
-            if (prefix := int(asn) // 65536) != 0:
-                return f"{prefix}.{int(asn) % 65536}"
-            return f"{int(asn) % 65536}"
-
-        if bgp_as_notation == "asplain" and "." in str(asn):
-            prefix, suffix = map(int, str(asn).split("."))
-            return str(int(prefix) * 65536 + int(suffix))
+        # TODO: This calculation would not work for -12345.12234 type of input.
+        # Need to handle in schema probably.
+        try:
+            if self.bgp_as_notation == "asdot" and "." not in str(asn):
+                if (prefix := int(asn) // 65536) != 0:
+                    return f"{prefix}.{int(asn) % 65536}"
+                return f"{int(asn) % 65536}"
+
+            if self.bgp_as_notation == "asplain" and "." in str(asn):
+                prefix, suffix = map(int, str(asn).split("."))
+                return str(int(prefix) * 65536 + int(suffix))
+        except ValueError as e:
+            msg = f"The value for 'asn' must be an integer or a float, but received '{asn}'."
```

Comment:
I don't think this is a correct error message given that you can very validly give a BGP asn of `1.65001` which is a string and not an int (and definitely not a float though I see why a developer would write this ;) ).

I would make two try catch inside the if blocks with:

```suggestion
            msg = f"Failed to convert '{asn}' to an integer when converting the BGP AS asdot notation`."
```

```suggestion
            msg = f"Failed to convert '{prefix}' or '{suffix} to an integer when converting the BGP AS '{asn}' to asplain notation`."
```

Reply:
Fixed